cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.relational;
+
+import java.util.Objects;
+
+/**
+ * A default implementation of Expression that simply stores the expression as a string.
+ * The string is assumed to be ANSI SQL compliant.
+ * It is the responsibility of the factory creating StringExpression objects to enforce correctness of the expression.
+ */
+public class StringExpression implements Expression {
+  private final String expression;
+
+  public StringExpression(String expression) {
+    this.expression = expression;
+  }
+
+  public String getExpression() {
+    return expression;
+  }
+
+  @Override
+  public boolean isValid() {
+    return true;
+  }
+
+  @Override
+  public String getValidationError() {
+    return "";
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof StringExpression)) {
+      return false;
+    }
+    StringExpression that = (StringExpression) o;
+    return getExpression().equals(that.getExpression());
```

Comment:
This equals method doesn't have null checks. 

I would check if `o` is null and use `Objects.equals()` to perform the equality comparison.

Reply:
Fixed this, although since the class doesn't have a default constructor I don't quite see how `getExpression()` will be null.